### ISSUE TITLE
Multiple issues with large number of RTs and RateLimitConfigs

### ISSUE THREAD

**Issue by bdecoste** [solo-io/gloo#7042]
### Gloo Edge Version

1.11.x

### Describe the bug

When I create a large number of RouteTables and RateLimitConfigs, I ran into several problems with the state:

1) Added a delegateAction to the VS referencing a non-existent RT (name: invalid). The VS was accepted, `glooctl check` passes, and the proxy resource was missing the invalid route. When I edited the missing RT name in the route to a different missing RT (name: invalid1), everything still passed. I then deleted the invalid route and  hit the next issue:
2) The VS status and `glooctl check` report that the RT referenced by the now deleted route is missing:
```
Error: 1 error occurred:
	* Found virtual service with warnings by 'gloo-system': gloo-system default (Reason: warning: 
  route table gloo-system.invalid1 missing)
```
Restarting the gateway&gloo pods did not resolve the issue. Deleting the proxy resource and restarting the gateway&gloo pods eventually resolved the issue.

3) Apply the invalid delegateAction. This will cause the validation to correctly fail. Then apply invalid-routetable.yaml (attached). This will cause the validation to correctly fail. Then apply valid-routetable.yaml and the validation incorrectly fails. 

### Steps to reproduce the bug

See attached 

### Expected Behavior

Config processed correctly 

### Additional Context

1.11.30

**Comment by bdecoste** [comment]
[create_rts.tar.gz](https://github.com/solo-io/gloo/files/9435243/create_rts.tar.gz)

**Comment by bdecoste** [comment]
[busted.tar.gz](https://github.com/solo-io/gloo/files/9435271/busted.tar.gz)

**Comment by bdecoste** [comment]
[rts.tar.gz](https://github.com/solo-io/gloo/files/9435285/rts.tar.gz)

**Comment by bdecoste** [comment]
Possibly seen in 1.12.9

**Comment by nfuden** [comment]
The new Rate limit caching may help with this

**Comment by nfuden** [comment]
Lets start with testing if this is still an issue in 1.12 given the work thats been going on in the Rate llimit space and see if there are gaps before we start backporting whole-sale.

**Comment by chrisgaun** [comment]
This is high priority item for customer. They are blocked.

**Comment by kdorosh** [comment]
We have validated that the issue is fixed on 1.12.

There are numerous changes that went into 1.12 that may have helped here; some can be backported and some cannot (e.g. pods merge). Can the customer upgrade to a version with the fix?
